File: sol/3025/2525.cpp
struct Solution {
    int numberOfPairs(vector<vector<int>>& points) {
        ranges::sort(points, [](auto &a, auto &b){
            return a[1] * 100 - a[0] < b[1] * 100 - b[0];
        });
        int n = points.size(), res = 0;
        for (int i = 0; i < n; ++i){
            for (int j = i + 1; j < n; ++j){
                if (points[i][0] < points[j][0]){
                    continue;
                }
                bool isValid = true;
                for (int k = i + 1; k < j; ++k){
                    if (points[i][0] >= points[k][0] && points[k][0] >= points[j][0]){
                        isValid = false;
                        break;
                    }
                }
                res += isValid;
            }
        }
        return res;
    }
};

File: sol/3025/2525.py
class Solution:
    def numberOfPairs(self, points: List[List[int]]) -> int:
        points.sort(key=lambda pt: (pt[1], -pt[0]))
        res = 0
        n = len(points)
        for i in range(n):
            for j in range(i + 1, n):
                if points[i][0] < points[j][0]:
                    continue
                for k in range(i + 1, j):
                    if points[i][0] >= points[k][0] >= points[j][0]:
                        break
                else:
                    res += 1
        return res
